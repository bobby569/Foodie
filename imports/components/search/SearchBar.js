import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Input } from 'antd';
const { Search } = Input;

class SearchBar extends Component {
	render() {
		return (
			<div className="search">
				<Search placeholder="Search" onSearch={val => console.log(val)} enterButton />
			</div>
		);
	}
}

export default SearchBar;
