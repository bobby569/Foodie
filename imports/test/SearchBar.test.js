import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchBar from '../components/search/SearchBar';

const onSearch = jest.fn();

describe('SearchBar', () => {
	const search_bar = shallow(<SearchBar onSearch={onSearch} />);
	const rendered = renderer.create(<SearchBar onSearch={onSearch} />);

	it('Renders correctly', () => {
		expect(search_bar.exists()).toBe(true);
		expect(rendered.toJSON()).toMatchSnapshot();
	});
});
