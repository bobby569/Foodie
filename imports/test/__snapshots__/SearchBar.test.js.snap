// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Renders correctly 1`] = `
<div
  className="search"
>
  <span
    className="ant-input-search ant-input-search-enter-button ant-input-affix-wrapper"
    style={undefined}
  >
    <input
      className="ant-input"
      disabled={false}
      onKeyDown={[Function]}
      placeholder="Search support by Edamam"
      size={undefined}
      style={null}
      type="text"
    />
    <span
      className="ant-input-suffix"
    >
      <button
        className="ant-btn ant-input-search-button ant-btn-primary"
        onClick={[Function]}
        type="button"
      >
        <i
          className="anticon anticon-search"
        />
      </button>
    </span>
  </span>
</div>
`;
